docs/coordinates/performance.inc.rst

```diff
@@ -56,3 +56,157 @@ properties::
   EXAMPLE END
+
+
+Improving Performance for Arrays of ``obstime``
+-----------------------------------------------
+
+The most expensive operations when transforming between observer-dependent coordinate
+frames (e.g. ``AltAz``) and sky-fixed frames (e.g. ``ICRS``) are the calculation
+of the orientation and position of Earth.
+
+If |skycoord| instances with a large ``obstime`` array are transformed,
+these calculations can be sped up by factors up to 100, whilst still keeping micro-arcsecond precision,
+by utilizing interpolation instead of calculating Earth orientation parameters for each individual point.
+
+This can be achieved through the ``erfa_astrom`` state and the ``ErfaAstromInterpolator``
+class like this
+
+..
+  EXAMPLE START
+  Improving performance for obstime arrays
+
+To use interpolation for the astrometric values in coordinate transformation, use::
+
+   >>> from astropy.coordinates import SkyCoord, EarthLocation, AltAz
+   >>> from astropy.coordinates.erfa_astrom import erfa_astrom, ErfaAstromInterpolator
+   >>> from astropy.time import Time
+   >>> from time import perf_counter
+   >>> import numpy as np
+   >>> import astropy.units as u
+
+
+   >>> # array with 10000 obstimes
+   >>> obstime = Time('2020-01-01T20:00') + np.linspace(0, 6, 10000) * u.hour
+   >>> location = location = EarthLocation(lon=-17.89 * u.deg, lat=28.76 * u.deg, height=2200 * u.m)
+   >>> frame = AltAz(obstime=obstime, location=location)  # doctest:+REMOTE_DATA
+   >>> crab = SkyCoord.from_name('Crab')  # doctest:+REMOTE_DATA
+
+   >>> # transform with default transformation and print duration
+   >>> t0 = perf_counter()
+   >>> crab_altaz = crab.transform_to(frame)  # doctest:+REMOTE_DATA +IGNORE_WARNINGS
+   >>> print(f'Transformation took {perf_counter() - t0:.2f} s')  # doctest:+ELLIPSIS
+   Transformation took ... s
+
+   >>> # transform with interpolating astrometric values
+   >>> t0 = perf_counter()
+   >>> with erfa_astrom.set(ErfaAstromInterpolator(300 * u.s)):  # doctest:+REMOTE_DATA
+   ...     crab_altaz_interpolated = crab.transform_to(frame)  # doctest:+REMOTE_DATA +IGNORE_WARNINGS
+   >>> print(f'Transformation took {perf_counter() - t0:.2f} s')  # doctest:+ELLIPSIS
+   Transformation took ... s
+
+   >>> err = crab_altaz.separation(crab_altaz_interpolated)  # doctest:+REMOTE_DATA
+   >>> print(f'Mean error of interpolation: {err.to(u.microarcsecond).mean():.4f}')  # doctest:+ELLIPSIS +REMOTE_DATA
+   Mean error of interpolation: 0.0167 uarcsec
+
+   >>> # To set erfa_astrom for a whole session, use it without context manager:
+   >>> erfa_astrom.set(ErfaAstromInterpolator(300 * u.s))  # doctest:+SKIP
+
+..
+  EXAMPLE END
+
+
+Here, we look into choosing an appropriate ``time_resolution``. 
+We will transform a single sky coordinate for lots of observation times from
+``ICRS`` to ``AltAz`` and evaluate precision and runtime for different values
+for ``time_resolution`` compared to the non-interpolating, default approach.
+
+.. plot::
```

Comment:
Very nice to include this plot - best possible way to help people determine how to check